### PR TITLE
Clear Inline Functions

### LINKED ISSUE #74: Clear inline functions
Add `clearInlineFunctions` to the API and default to false. Use regex to detect if an attribute's value matches the pattern of a function, then replace the value with a placeholder like `[function]`.

Spinoff of #71.

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
[Coverage Status](https://coveralls.io/github/tjw-lint/jest-serializer-vue-tjw?branch=clear-fn)

* [x] Functionality
* [x] Tests
* [x] Documentation
* [x] Version bump
* [x] ~~Update deps~~

### CODE REVIEW

File: src/stringManipulation.js [hunk 1]
```diff
@@ -93,6 +93,34 @@ function clearAttributes (html, options) {
...
+  if (
+    options &&
+    options.clearInlineFunctions
+  ) {
+    const $ = helpers.$(html);
+
+    $('*').each(function (index, element) {
+      Object.keys(element.attribs).forEach(function (attributeName) {
+        let value = element.attribs[attributeName].trim();
+        if (value.startsWith('function') && value.endsWith('}')) {
```
Comment (TheJaredWilcurt) [inline src/stringManipulation.js:113]: this `if` statement needs replaced with a more robust regex pattern

Comment (TheJaredWilcurt) [inline src/stringManipulation.js:113]:
```suggestion
        let pattern = /function *\(([a-zA-Z0-9_$],* *){0,}\) *{(.*\n*)*}/;
        if (pattern.test(value)) {
```

**Comment by cameron-martin** [comment]
There's a bunch of cases where this will break and I don't think this is the right approach personally, but otherwise it gets the job done in most cases I guess.

**Comment by TheJaredWilcurt** [comment]
@cameron-martin can you elaborate with examples. I have never used inline functions, so I'm trying to create this feature for those that do. That's why I'm asking for your input.

**Comment by cameron-martin** [comment]
Cases where this will break that I can see are:

* Where the prop is actually a string that happens to contain something that matches the regex.
* Pretty much all the cases where you do ` *` need to take into account other whitespace.
* Identifiers in JS can include way more than just `[a-zA-Z0-9_$]`.
* Unless I'm missing something, it assumes transpilation of arrow functions, since the regex doesn't match arrow functions. This may not be the case.

The last ones are fixable by changing the regex, but the first one isn't, although may not be much of an issue in practice.

File: tests/__snapshots__/InlineFunctions.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<div>
+  <h1 title="function () {
+        return true;
+      }">Classic</h1>
+  <h2 title="function () {
+        return true;
+      }">Multi-line</h2>
+  <h3 title="function (x) {
+        return !x;
+      }">Arrow</h3>
```
Comment (TheJaredWilcurt) [inline tests/__snapshots__/InlineFunctions.test.js.snap:13]: @cameron-martin You can see here what Vue renders the templates out to. the One liner, Multi-line, and Arrow function, are all output as the same multi-line approach.

Comment (cameron-martin) [reply tests/__snapshots__/InlineFunctions.test.js.snap:13]: Interesting. How does that transformation get peformed?

**Comment by TheJaredWilcurt** [comment]
@cameron-martin 

>  Where the prop is actually a string that happens to contain something that matches the regex.

`<div title="function stuff () { whatever; }">` yes, this would get converted to `<div title="[function]">`, but this is an opt-in feature. You need to set `clearInlineFunctions: true` in the API. So this should catch anyone off-guard.

> Pretty much all the cases where you do * need to take into account other whitespace.
> Identifiers in JS can include way more than just [a-zA-Z0-9_$].

I've updated the regex and added in test cases for it. Though I am very open to PR suggestions for improvements to the regex or tests around the regex.

> Unless I'm missing something, it assumes transpilation of arrow functions, since the regex doesn't match arrow functions. This may not be the case.

Yes, Vue converts all inline functions to look the same by the time the regex checks it.

File: tests/__snapshots__/InlineFunctions.test.js.snap (hunk 2 above)
Comment (TheJaredWilcurt) [reply tests/__snapshots__/InlineFunctions.test.js.snap:13]: Not sure when or how, but by the time we are checking for it, they are all the same.

**Comment by cameron-martin** [comment]
Looks good to me, then.

**Pull request merged**